### PR TITLE
Support for new XML format

### PR DESCRIPTION
**Pull request by snukky**
This PR adds support for the new XML format replacing the SGML format, see https://github.com/wmt-conference/wmt-format-tools

Changes proposed in the pull request:
- Added the new XML format associated with the extension `.xml` for test sets and submissions.
- BLEU/Chrf scores are computed on the first reference only.
- Documents from test suites are ignored when calculating BLEU/ChrF scores.
- XML schema is validated upon submission.
- Added unit tests covering the above.

@AppraiseDev/core-team

### CODE REVIEW

**Comment by cfedermann** [comment]
Thank you, @snukky.

Will look into this shortly 😊

File: leaderboard/tests.py [hunk 1]
```diff
@@ -15,11 +16,60 @@
 from leaderboard.models import Team
 from leaderboard.models import TestSet
 from leaderboard.models import TEXT_FILE
+from leaderboard.models import XML_FILE
+from leaderboard.utils import analyze_xml_file
+from leaderboard.utils import MISSING_TRANSLATION_MESSAGE
+from leaderboard.utils import process_xml_to_text
 from ocelot.settings import BASE_DIR
 
 TESTDATA_DIR = os.path.join(BASE_DIR, 'leaderboard/testdata')
```
Comment (cfedermann) [inline leaderboard/tests.py:25]: Use `pathlib.Path` object instead?

File: leaderboard/models.py [hunk 2]
```diff
@@ -247,22 +424,24 @@ class TestSet(models.Model):
...
         max_length=4,
     )
 
     src_file = models.FileField(
         blank=True,
         upload_to='testsets',
-        help_text='SGML or text file containing test set source',
+        help_text='SGML, XML or text file containing test set source',
         null=True,
+        validators=[validate_xml_src_testset],
```
Comment (cfedermann) [inline leaderboard/models.py:436]: Validation here will enforce that `file_format` is `XML`. This makes the `file_format` field above redundant.

Should we add a validation method which considers `file_format`?

File: leaderboard/models.py [hunk 3]
```diff
@@ -756,20 +988,18 @@ def _compute_score(self):
         ref_text_path = self.get_ref_text(path_only=True)
 
         try:
-            hyp_stream = (x for x in open(hyp_text_path, encoding='utf-8'))
-            ref_stream = (r for r in open(ref_text_path, encoding='utf-8'))
+            hyp_stream = [x for x in open(hyp_text_path, encoding='utf-8')]
```
Comment (cfedermann) [inline leaderboard/models.py:991]: @snukky: just curious, why change to list comprehension?

**Review by cfedermann — APPROVED** [review]
Thank you very much, @snukky.

Added a few comments, but no show stoppers. Please go ahead with the merge.

File: leaderboard/tests.py (hunk 1 above)
Comment (snukky) [reply leaderboard/tests.py:25]: Yes, we even have a task opened for that. For now, as this is only in unit tests, I've ignored this in order to save some time. I can change it tomorrow.

File: leaderboard/models.py (hunk 2 above)
Comment (snukky) [reply leaderboard/models.py:436]: I think this shouldn't enforce the format thanks to:
https://github.com/AppraiseDev/OCELoT/blob/c241adb715b6576422a503ee9f23f90233756a79/leaderboard/models.py#L213-L214
Or I simply didn't get the question :)

File: leaderboard/models.py (hunk 3 above)
Comment (snukky) [reply leaderboard/models.py:991]: Hm. First, I've changed it to lists to be able to print their values when debugging. Then I decided to leave them for the sake of removing two duplicated lines below. But I haven't thought much if there are other potential issues from this change, e.g. optimization-wise.

**Pull request merged**